src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -629,6 +626,418 @@ CreateOffer::cross (
+            std::stringstream txIdSs;
+            txIdSs << ".  tx: " << ctx.tx.getTransactionID();
+            auto txID = txIdSs.str();
+
+            // Sometimes one version deletes offers that the other doesn't
+            // delete.  That's okay, but keep track of it.
+            c = SBoxCmp::offerDelDiff;
+            s = kWarning;
+            int sides = diff.rmLhsDeletedOffers() ? 1 : 0;
+            sides |= diff.rmRhsDeletedOffers() ? 2 : 0;
+            if (!diff.hasDiff())
+            {
+                char const* type = "";
+                switch (sides)
+                {
+                case 1: type = "; V1 deleted more offers"; break;
+                case 2: type = "; V2 deleted more offers"; break;
+                case 3: type = "; V1 and V2 deleted different offers"; break;
+                default: break;
+                }
+                diffDesc = std::string(type) + txID;
+            }
+            else
+            {
+                // A difference without a broad classification...
+                c = SBoxCmp::diff;
+                std::stringstream ss;
+                ss << "; common entries: " << diff.commonCount()
+                    << "; V1 unique: " << diff.lhsOnlyCount()
+                    << "; V2 unique: " << diff.rhsOnlyCount() << txID;
+                diffDesc = ss.str();
+            }
+        }
+        j.stream (s) << "OfferCrossV2: " << name << " different" << diffDesc;
+    }
+    return c;
+}
+
+std::pair<TER, Amounts>
+CreateOffer::cross (
+    PaymentSandbox& psb,
+    PaymentSandbox& psbCancel,
+    Amounts const& takerAmount)
+{
+    // There are amendments for V2 offer crossing and for comparing results
+    // between V1 and V2 offer crossing.  Turn those into bools.
+    bool const useV2 {
+        psb.rules().enabled (
+            featureOfferCrossV2, ctx_.app.config().features) };
+
+    bool const doCompare {
+        psb.rules().enabled (
+            featureCompareOfferCrossV1V2, ctx_.app.config().features) };
+
+    if (doCompare)
+    {
+        // If we're comparing then there's a good chance we're replaying
+        // a transaction.  Make it easy to see the transaction ID in the
+        // debugger.
+        std::stringstream ss;
+        ss << ctx_.tx.getTransactionID();
+        auto const txID = ss.str();
+        char const* txIDptr = txID.c_str();
+        (void) txIDptr;
+    }
+
+    // Decide which version of offer crossing to run.  Run both if comparing.
+    using CallCrossPtr = std::pair<TER, Amounts>
+        (CreateOffer::*) (PaymentSandbox&, PaymentSandbox&, Amounts const&);
+
+    auto callCross =
+        [this] (bool use, CallCrossPtr callPtr, PaymentSandbox& psb,
+            PaymentSandbox& psbCancel, Amounts const& takerAmount)
+        {
+            if (use)
+                return ((this)->*callPtr)(psb, psbCancel, takerAmount);
+            return std::make_pair (tecINTERNAL, takerAmount);
+        };
```

Comment:
I was thinking something like this:
```
    auto const v1r = (useV2 && !doCompare)
        ? std::make_pair(tecINTERNAL, takerAmount)
        : crossV1(psb, psbCancel, takerAmount);
```
If we did that, it'd be shorter (6 loc vs. 14 loc), and easier to understand too (no need to refer to the lambda, no pointers to members, no typedef for the pointers to members).

It's a nit for sure, you're call.